### PR TITLE
OSOE-49: Loop refactoring

### PR DESCRIPTION
**Pull request by 0liver**
During review of OSOE-49.

### CODE REVIEW

File: Lombiq.DataTables/Tests/Lombiq.DataTables.Tests/UnitTests/Services/ExportTests.cs [hunk 1]
```diff
@@ -56,20 +56,14 @@ public class ExportTests
             mocker => mocker.MockStringLocalizer<ExcelDataTableExportService>());
 
         Dictionary<int, string> customNumberFormat = null;
-        int columnIndex = 0;
-        do
+        int columnIndex;
+        for (columnIndex = 0; columnIndex < columns.Length && columns[columnIndex - 1].Name != "Time"; columnIndex++)
```
Comment (sarahelsaig) [inline Lombiq.DataTables/Tests/Lombiq.DataTables.Tests/UnitTests/Services/ExportTests.cs:60]: This is not an equivalent transformation, you've turned it into a while loop. Unlike those, a do..while loop always runs at least once. You could rewrite the condition to `columnIndex == 0 || (...)` but that's kind of ugly and it obfuscates a language feature that do..while explicitly shows.

**Review by sarahelsaig — CHANGES_REQUESTED** [review]

File: Lombiq.DataTables/Tests/Lombiq.DataTables.Tests/UnitTests/Services/ExportTests.cs [hunk 2]
```diff
@@ -56,20 +56,14 @@ public class ExportTests
...
         {
             if (columns[columnIndex].Name == "Time")
             {
-                customNumberFormat = new Dictionary<int, string>
-                {
-                    [columnIndex + 1] = "h:mm:ss AM/PM",
-                };
+                customNumberFormat = new Dictionary<int, string> { [columnIndex + 1] = "h:mm:ss AM/PM" };
```
Comment (sarahelsaig) [inline Lombiq.DataTables/Tests/Lombiq.DataTables.Tests/UnitTests/Services/ExportTests.cs:64]: This is nice, I went ahead and integrated it.

File: Lombiq.DataTables/Tests/Lombiq.DataTables.Tests/UnitTests/Services/ExportTests.cs (hunk 1 above)
Comment (0liver) [reply Lombiq.DataTables/Tests/Lombiq.DataTables.Tests/UnitTests/Services/ExportTests.cs:60]: I'm pretty sure the first iteration will always run (for `i == 0`), unless the length of `columns` is `0`, in which case the `do` statement would actually throw up.

Comment (sarahelsaig) [reply Lombiq.DataTables/Tests/Lombiq.DataTables.Tests/UnitTests/Services/ExportTests.cs:60]: Have you tried it? If i switch to your branch all ExportTests fail.

```
System.IndexOutOfRangeException
Index was outside the bounds of the array.
   at Lombiq.DataTables.Tests.UnitTests.Services.ExportTests.ExportTableShouldMatchExpectation(String note, Object[][] dataSet, ValueTuple`3[] columns, String[][] pattern, Int32 start, Int32 length, Int32 orderColumnIndex) in E:\aux_c\repositories\orchard-core\Open-Source-Orchard-Core-Extensions2\src\Modules\Lombiq.DataTables\Lombiq.DataTables\Tests\Lombiq.DataTables.Tests\UnitTests\Services\ExportTests.cs:line 60
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 264
--- End of stack trace from previous location ---
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 48
   at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\Dev\xunit\xunit\src\xunit.core\Sdk\ExceptionAggregator.cs:line 90
```

Comment (0liver) [reply Lombiq.DataTables/Tests/Lombiq.DataTables.Tests/UnitTests/Services/ExportTests.cs:60]: I didn't, but now I can see it: `columnIndex-1` will be `-1` in the first loop run. Ouch. Then please, reject my suggestion.

Comment (0liver) [reply Lombiq.DataTables/Tests/Lombiq.DataTables.Tests/UnitTests/Services/ExportTests.cs:60]: I mean, I'll close the PR.

**Comment by 0liver** [comment]
Loop refactoring failed because the logic changed.

**Pull request closed without merging**